src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -190,14 +190,26 @@ void erase(T port) {
 }  // namespace ric_attr
 
 namespace init {
+
+namespace {
+
+void add_node_with_inputs_to_vector(const std::shared_ptr<ov::Node>& node, NodeVector& vector) {
+    vector.push_back(node);
+    const auto& inputs = node->inputs();
+    for (auto it = inputs.begin(); it != inputs.end(); ++it) {
```

Comment:
Detail:
```suggestion
    for (const auto& input = inputs) {
        vector.push_back(input->get_source_output().get_node_shared_ptr());
    }
```
or use `for_each` ?

Reply:
Fixed 😊